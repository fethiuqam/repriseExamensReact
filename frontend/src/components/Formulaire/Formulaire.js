import '../../styles/StyleEtudiant.css'
import SectionFormulaire from "../SectionFormulaire/SectionFormulaire"
import { TextField, InputLabel, Select, Container } from "@material-ui/core/";
import { Grid, Typography } from "@mui/material";
import { useForm, Controller } from "react-hook-form";
import Button from '../BoutonFormulaire/BoutonFormulaire';
import MiseEnPage from '../MiseEnPage/MiseEnPage';
import { useState, useEffect, useContext } from "react";
import { useNavigate } from 'react-router-dom';
import * as React from "react";
import Banniere from '../Banniere/Banniere';
import AuthContext from "../../context/AuthProvider";
import Box from '@mui/material/Box';
import AttachmentIcon from '@mui/icons-material/Attachment';
import DeleteIcon from '@mui/icons-material/Delete';
import { FileUploader } from "react-drag-drop-files";




// stocke les infos des etudiant TODO : connecter avec les identifiants
const infosEtudiant = {
    nom: "Tremblay",
    prenom: "Jeanne",
    codePermanent: "TREJ90021580"
}

const Formulaire = () => {
    const {jwt} = useContext(AuthContext);
    const {
        register,
        handleSubmit,
        formState: { errors },
        getValues,
        control
    } = useForm();

    let formData = new FormData();

    // states pour la gestion des pieces jointes
    const fileTypes = ["JPEG", "PNG", "PDF", "JPG"];
    const [files, setFiles] = useState([]);

    /***
     * Ajoute des nouveaux fichiers
     * @param {*} nouveauFichiers les nouveaux fichiers a ajouter
     */
    const handleChangeFile = (nouveauFichiers) => {
        setFiles(ancienFichiers => [...ancienFichiers, ...nouveauFichiers]);
    };

    /**
     * Fonction appelee aux changements effectues aux fichiers
     * Notes: le commentaire eslint-disable-line etait la methode
     * la plus efficace pour enlever un warning de dependance qui
     * n'avait pas d'impact sinon
     */
    useEffect(() => {
        ajouterFilesFormdata();
    }, [files]); // eslint-disable-line

    /**
     * Fonction pour supprimer un fichier donne 
     * @param {*} nombre index du fichier a enlever
     */
    const supprimerFichier = (nombre) => {
        try {
            const nouvArray = [...files];
            nouvArray.splice(nombre, 1);
            setFiles(nouvArray);
        } catch (e){
            console.error("Erreur dans la supression fichier")
        }
    };

    /***
     * Fonction pour faire un update des fichiers dans formdata
     * a partir des fichiers presents dans le state files
     */
    const ajouterFilesFormdata = () => {
        formData.delete("files");
        files.forEach(f => formData.append("files", f));
    }

    const [isSubmitting, setIsSubmitting] = useState(false);
    const navigate = useNavigate();
    const [error, setError] = useState();

    async function handleFormSubmit(data) {

        try {
            setIsSubmitting(true);

            formData.append('nouvelleDemande',
                new Blob([JSON.stringify(data)], { type: "application/json" }));
            
            var response = await fetch("/api/demandes", {
                method: "POST",
                body: formData,
                headers: { 'Authorization':'Bearer ' + jwt }
            });

            if (!response.ok) {
                throw new Error("Erreur du POST");
            }

            navigate("/");
        } catch (error) {
            setError(error);
            setIsSubmitting(false);
        } 
    }

    const onSubmit = (data) => {
        handleFormSubmit(data);
    };


    return (
        <>
            <MiseEnPage titre={'Demande de reprise d\'examen'}
                children={
                    <form onSubmit={handleSubmit(onSubmit)}>
                        {error && 
                            <Banniere 
                                id="banniereErreur" 
                                texte="ERREUR : Votre demande n'a pas pu être envoyée."
                                className="banniereErreur"/>
                        }
                        <SectionFormulaire
                            title={'INFORMATIONS PERSONNELLES'}
                            children={
                                <>
                                    <table className="affichageInfosTable">
                                        <tbody>
                                            <tr>
                                                <th>Nom</th>
                                                <th>{infosEtudiant.nom}</th>
                                            </tr>
                                            <tr>
                                                <th>Prénom</th>
                                                <th>{infosEtudiant.prenom}</th>
                                            </tr>
                                            <tr>
                                                <th>Code permanent</th>
                                                <th>{infosEtudiant.codePermanent}</th>
                                            </tr>
                                        </tbody>
                                    </table>
                                </>
                            }
                        />
                        <SectionFormulaire
                            title={'INFORMATIONS SUR L\'ABSENCE'}
                            children={
                                <>
                                    <Grid container spacing={2} className="affichageInfosTable">
                                        <Grid item xs={3}>
                                            <Grid container>
                                                <Grid item xs={4}>
                                                    <InputLabel id="etiquette-signelCours">Sigle du
                                                        cours</InputLabel>
                                                </Grid>
                                                <Grid item xs={8}>
                                                    <Select
                                                        data-testid="sigleCoursTestId"
                                                        native
                                                        defaultValue=""
                                                        className="inputStandard"
                                                        labelId="etiquette-signelCours"
                                                        id="sigleCours-input"
                                                        name="sigleCours"
                                                        label="Sigle du cours"
                                                        style={{
                                                            width: "90%"
                                                        }}
                                                        {...register("sigleCours", { required: true })}
                                                        error={Boolean(errors.sigleCours)}
                                                    >
                                                        <option name={'INM5151'} value={'INM5151'}>INM5151</option>
                                                        <option value={'INF6150'}>INF6150</option>
                                                        <option value={'INF5171'}>INF5171</option>
                                                    </Select>
                                                </Grid>
                                            </Grid>
                                        </Grid>
                                        <Grid item xs={2}>
                                            <Grid container>
                                                <Grid item xs={4}>
                                                    <InputLabel id="etiquette-groupeCours">Groupe</InputLabel>
                                                </Grid>
                                                <Grid item xs={8}>
                                                    <TextField // TODO va être rempli automatiquement quand on choisit le sigle
                                                        data-testid="groupeCoursTestId"
                                                        defaultValue=""
                                                        className="inputStandard"
                                                        labelid="etiquette-groupeCours"
                                                        id="groupeCours-input"
                                                        name="groupeCours"
                                                        label="Groupe"
                                                        style={{
                                                            width: "90%"
                                                        }}
                                                        {...register("groupeCours", { required: true })}
                                                        error={Boolean(errors.groupeCours)}
                                                    />
                                                </Grid>
                                            </Grid>
                                        </Grid>
                                        <Grid item xs={3}>
                                            <Grid container>
                                                <Grid item xs={4}>
                                                    <InputLabel id="etiquette-enseignant">Enseignant</InputLabel>
                                                </Grid>
                                                <Grid item xs={8}>
                                                    <TextField // TODO va être rempli automatiquement quand on choisit le sigle
                                                        data-testid="enseignantTestId"
                                                        defaultValue=""
                                                        className="inputStandard"
                                                        labelid="etiquette-enseignant"
                                                        id="enseignant-input"
                                                        name="enseignant"
                                                        label="Enseignant"
                                                        style={{
                                                            width: "90%"
                                                        }}
                                                        {...register("enseignant", { required: true })}
                                                        error={Boolean(errors.enseignant)}
                                                    />
                                                </Grid>
                                            </Grid>
                                        </Grid>
                                        <Grid item xs={3}></Grid>
                                    </Grid>
                                    <Controller control={control}
                                        name="dateDebutEtFinAbsence"
                                        render={({ field: { onChange, value } }) => {
                                            return (
                                                <Grid container spacing={2} className="affichageInfosTable">
                                                    <Grid item xs={3}>
                                                        <TextField
                                                            id="dateDebut"
                                                            label="Début de l'absence"
                                                            name="absenceDateDebut"
                                                            type="date"
                                                            fullWidth
                                                            InputProps={{
                                                                inputProps: {
                                                                    min: new Date(new Date().getFullYear(), new Date().getMonth() - 3, new Date().getDate()).toISOString().slice(0, 10),
                                                                    max: getValues("absenceDateFin")
                                                                }
                                                            }}
                                                            InputLabelProps={{
                                                                shrink: true,
                                                            }}
                                                            defaultValue={new Date().toISOString().slice(0, 10)}
                                                            {...register("absenceDateDebut", {
                                                                required: "Ce champs est requis.",
                                                                onChange: onChange
                                                            })}
                                                            error={Boolean(errors.absenceDateDebut)}
                                                        />
                                                    </Grid>
                                                    <Grid item xs={3}>
                                                        <TextField
                                                            id="dateFin"
                                                            label="Fin de l'absence"
                                                            name="absenceDateFin"
                                                            type="date"
                                                            fullWidth
                                                            InputProps={{
                                                                inputProps: {
                                                                    min: getValues("absenceDateDebut"),
                                                                    max: new Date().toISOString().slice(0, 10)
                                                                }
                                                            }}
                                                            InputLabelProps={{
                                                                shrink: true,
                                                            }}
                                                            defaultValue=""
                                                            {...register("absenceDateFin", {
                                                                required: "Ce champs est requis.",
                                                                onChange: onChange
                                                            })}
                                                            error={Boolean(errors.absenceDateFin)}
                                                        />
                                                    </Grid>
                                                </Grid>
                                            )
                                        }} />
                                    <br />
                                </>
                            }
                        />
                        <SectionFormulaire
                            title={'JUSTIFICATION DE L\'ABSENCE'}
                            children={
                                <div className="sectionFormulaireDivContenuRangee">
                                    <div>
                                        <InputLabel id="etiquette-motif-absence">Motif</InputLabel>
                                        <Select
                                            data-testid="motifTestId"
                                            native
                                            defaultValue=""
                                            className="inputStandard"
                                            labelId="etiquette-motif-absence"
                                            id="menu-motif-absence"
                                            name="motifAbsence"
                                            label="Motif"
                                            {...register("motifAbsence", { required: true })}
                                            error={Boolean(errors.motifAbsence)}
                                        >
                                            <option value={'MEDICAL'}>Hospitalisation</option>
                                            <option value={'DECES'}>Décès</option>
                                            <option value={'ACCIDENT'}>Accident</option>
                                            <option value={'JUDICIAIRE'}>Convocation à un tribunal</option>
                                            <option value={'RELIGIEUX'}>Motif religieux</option>
                                            <option value={'AUTRE'}>Autre</option>
                                        </Select>
                                    </div>
                                    <div>
                                        <TextField
                                            id="explications-input"
                                            name="absenceDetails"
                                            label="Explications détaillées"
                                            type="text"
                                            multiline
                                            fullWidth
                                            minRows={4}
                                            variant="outlined"
                                            {...register("absenceDetails", { required: "Ce champs est requis." })}
                                            error={Boolean(errors.absenceDetails)}
                                        />
                                    </div>
                                    <div className="textColor">
                                        <Typography>Justificatifs</Typography>
                                    </div>
                                    <div>Types de fichiers acceptés : JPEG, PNG et PDF </div>
                                    <div>
                                        <FileUploader
                                            multiple={true}
                                            handleChange={handleChangeFile}
                                            onTypeError={(err) => console.log("Erreur type fichier")}
                                            name="file"
                                            types={fileTypes}
                                            label="Deposez vos fichiers ici"
                                            fileOrFiles={null}
                                            children={
                                            <Box component="div" sx={{
                                                width: 800,
                                                height: 200,
                                                p: 1,
                                                border: '1px dashed grey',
                                                '&:hover': {
                                                    backgroundColor: 'primary.main',
                                                    opacity: [0.6, 0.5, 0.4],
                                                  },}}>
                                                <div className="center">
                                                    <h3>Glisser les pièces jointes pour ajouter des preuves justificatives</h3>
                                                </div>                                          
                                            </Box>
                                        }
                                        />
                                    </div>
                                    <div>
                                        <ul className="afficherFichiers">
                                            {files.map(function(file, idx){
                                                return (
                                                <li key={idx}>
                                                    <AttachmentIcon></AttachmentIcon>
                                                    {file.name}
                                                    <span className='divSupprimerFichierButton'>
                                                        <Button
                                                            className="buttonSupprimerFichier"
                                                            startIcon={<DeleteIcon/>}
                                                            disableElevation
                                                            color="error"
                                                            size="small"
                                                            onClick={() => {
                                                                supprimerFichier(idx);
                                                            }}
                                                        >
                                                        </Button>
                                                    </span>
                                                    <br/>
                                                </li>)
                                            })}
                                        </ul>
                                    </div>
                                </div>
                            }
                        />
                        <Container>
                            <div className="boutonsFormulaire">
                                <Button
                                    variant="outlined"
                                    onClick={() => {
                                        navigate("/");
                                    }}
                                >
                                    Annuler
                                </Button>
                                <Button
                                    id="boutonSoumettre"
                                    variant="contained"
                                    disableElevation
                                    type="submit"
                                    disabled={isSubmitting}
                                >
                                    Soumettre
                                </Button>
                            </div>
                        </Container>
                    </form>
                }
            />
        </>
    );
}

export default Formulaire;
