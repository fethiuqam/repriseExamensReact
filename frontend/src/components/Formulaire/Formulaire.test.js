import '@testing-library/jest-dom'
import {render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import Formulaire from "./Formulaire";
import { BrowserRouter } from 'react-router-dom'; 

// test pour vérifier si le formulaire comprend toute l'information requise

test("contient tous les champs requis", () => {
    
    render(
      <BrowserRouter>
        <Formulaire />
      </BrowserRouter>
    );

    const listeChamps = [
        "Prénom", 
        "Nom", 
        "Code permanent",
        "Sigle du cours",
        "Groupe",
        "Enseignant",
        "Début de l'absence", 
        "Fin de l'absence",
        "Motif",
        "Explications détaillées",
        "Justificatifs"
    ];

    listeChamps.forEach( (elem) => {
        expect(screen.getAllByText(elem)).toBeTruthy();
    })

});

/* test pour vérifier que le UI de création de DRE n'a pas changé
 Pour mettre à jour le snapshot, ajouter "-u" au script "test:ci", directement dans package.json. 
 ATTENTION : N'oubliez pas d'enlever le "-u" une fois le snapshot mis à jour, sinon il va juste se 
 mettre à jour chaque fois que les tests roulent, sans jamais comparer l'état actuel au snapshot enregistré. 
 */

test('Formulaire affiché correctement', () => {

  jest.useFakeTimers().setSystemTime(new Date('2020-01-01'));

  const FormulaireAvecRoute = () => {
    return(
      <BrowserRouter>
        <Formulaire />
      </BrowserRouter>
    );
  }

  const tree = TestRenderer
      .create(<FormulaireAvecRoute/>, {
        createNodeMock: () => document.createElement('textarea')
      })
      .toJSON();
    expect(tree).toMatchSnapshot();
});

