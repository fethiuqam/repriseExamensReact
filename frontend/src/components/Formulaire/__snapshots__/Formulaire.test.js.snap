// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formulaire affiché correctement 1`] = `
<div
  className="miseEnPage"
>
  <div
    className="corpsPage"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <h4
        className="MuiTypography-root titre MuiTypography-h4"
      >
        Demande de reprise d'examen
      </h4>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiContainer-root"
      >
        <div
          className="MuiPaper-root sectionFormulaire MuiPaper-outlined"
        >
          <div
            className="sectionFormulaireDivTitre"
          >
            <h3
              className="sectionFormulaireTitre"
            >
              INFORMATIONS PERSONNELLES
            </h3>
          </div>
          <div
            className="sectionFormulaireDivContenu"
          >
            <table
              className="affichageInfosTable"
            >
              <tbody>
                <tr>
                  <th>
                    Nom
                  </th>
                  <th>
                    Tremblay
                  </th>
                </tr>
                <tr>
                  <th>
                    Prénom
                  </th>
                  <th>
                    Jeanne
                  </th>
                </tr>
                <tr>
                  <th>
                    Code permanent
                  </th>
                  <th>
                    TREJ90021580
                  </th>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        className="MuiContainer-root"
      >
        <div
          className="MuiPaper-root sectionFormulaire MuiPaper-outlined"
        >
          <div
            className="sectionFormulaireDivTitre"
          >
            <h3
              className="sectionFormulaireTitre"
            >
              INFORMATIONS SUR L'ABSENCE
            </h3>
          </div>
          <div
            className="sectionFormulaireDivContenu"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 affichageInfosTable css-mhc70k-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                  >
                    <label
                      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
                      id="etiquette-signelCours"
                    >
                      Sigle du cours
                    </label>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
                  >
                    <div
                      className="MuiInputBase-root MuiInput-root MuiInput-underline inputStandard"
                      data-testid="sigleCoursTestId"
                      onClick={[Function]}
                      style={
                        Object {
                          "width": "90%",
                        }
                      }
                    >
                      <select
                        aria-invalid={false}
                        className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
                        defaultValue=""
                        id="sigleCours-input"
                        multiple={false}
                        name="sigleCours"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                      >
                        <option
                          name="INM5151"
                          value="INM5151"
                        >
                          INM5151
                        </option>
                        <option
                          value="INF6150"
                        >
                          INF6150
                        </option>
                        <option
                          value="INF5171"
                        >
                          INF5171
                        </option>
                      </select>
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSelect-icon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
              >
                <div
                  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                  >
                    <label
                      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
                      id="etiquette-groupeCours"
                    >
                      Groupe
                    </label>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
                  >
                    <div
                      className="MuiFormControl-root MuiTextField-root inputStandard"
                      data-testid="groupeCoursTestId"
                      labelid="etiquette-groupeCours"
                      style={
                        Object {
                          "width": "90%",
                        }
                      }
                    >
                      <label
                        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                        data-shrink={false}
                        htmlFor="groupeCours-input"
                        id="groupeCours-input-label"
                      >
                        Groupe
                      </label>
                      <div
                        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                        onClick={[Function]}
                      >
                        <input
                          aria-invalid={false}
                          autoFocus={false}
                          className="MuiInputBase-input MuiInput-input"
                          defaultValue=""
                          disabled={false}
                          id="groupeCours-input"
                          name="groupeCours"
                          onAnimationStart={[Function]}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          type="text"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                  >
                    <label
                      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
                      id="etiquette-enseignant"
                    >
                      Enseignant
                    </label>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
                  >
                    <div
                      className="MuiFormControl-root MuiTextField-root inputStandard"
                      data-testid="enseignantTestId"
                      labelid="etiquette-enseignant"
                      style={
                        Object {
                          "width": "90%",
                        }
                      }
                    >
                      <label
                        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                        data-shrink={false}
                        htmlFor="enseignant-input"
                        id="enseignant-input-label"
                      >
                        Enseignant
                      </label>
                      <div
                        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                        onClick={[Function]}
                      >
                        <input
                          aria-invalid={false}
                          autoFocus={false}
                          className="MuiInputBase-input MuiInput-input"
                          defaultValue=""
                          disabled={false}
                          id="enseignant-input"
                          name="enseignant"
                          onAnimationStart={[Function]}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          type="text"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 affichageInfosTable css-mhc70k-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                    data-shrink={true}
                    htmlFor="dateDebut"
                    id="dateDebut-label"
                  >
                    Début de l'absence
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiInput-input"
                      defaultValue="2020-01-01"
                      disabled={false}
                      id="dateDebut"
                      min="2019-10-01"
                      name="absenceDateDebut"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="date"
                    />
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                    data-shrink={true}
                    htmlFor="dateFin"
                    id="dateFin-label"
                  >
                    Fin de l'absence
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiInput-input"
                      defaultValue=""
                      disabled={false}
                      id="dateFin"
                      max="2020-01-01"
                      name="absenceDateFin"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="date"
                    />
                  </div>
                </div>
              </div>
            </div>
            <br />
          </div>
        </div>
      </div>
      <div
        className="MuiContainer-root"
      >
        <div
          className="MuiPaper-root sectionFormulaire MuiPaper-outlined"
        >
          <div
            className="sectionFormulaireDivTitre"
          >
            <h3
              className="sectionFormulaireTitre"
            >
              JUSTIFICATION DE L'ABSENCE
            </h3>
          </div>
          <div
            className="sectionFormulaireDivContenu"
          >
            <div
              className="sectionFormulaireDivContenuRangee"
            >
              <div>
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
                  id="etiquette-motif-absence"
                >
                  Motif
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline inputStandard"
                  data-testid="motifTestId"
                  onClick={[Function]}
                >
                  <select
                    aria-invalid={false}
                    className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
                    defaultValue=""
                    id="menu-motif-absence"
                    multiple={false}
                    name="motifAbsence"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                  >
                    <option
                      value="MEDICAL"
                    >
                      Hospitalisation
                    </option>
                    <option
                      value="DECES"
                    >
                      Décès
                    </option>
                    <option
                      value="ACCIDENT"
                    >
                      Accident
                    </option>
                    <option
                      value="JUDICIAIRE"
                    >
                      Convocation à un tribunal
                    </option>
                    <option
                      value="RELIGIEUX"
                    >
                      Motif religieux
                    </option>
                    <option
                      value="AUTRE"
                    >
                      Autre
                    </option>
                  </select>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSelect-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
              <div>
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                    data-shrink={false}
                    htmlFor="explications-input"
                    id="explications-input-label"
                  >
                    Explications détaillées
                  </label>
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                    onClick={[Function]}
                  >
                    <textarea
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                      disabled={false}
                      id="explications-input"
                      name="absenceDetails"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      rows={4}
                      style={
                        Object {
                          "height": -4,
                          "overflow": "hidden",
                        }
                      }
                    />
                    <textarea
                      aria-hidden={true}
                      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                      readOnly={true}
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "hidden",
                          "position": "absolute",
                          "top": 0,
                          "transform": "translateZ(0)",
                          "visibility": "hidden",
                        }
                      }
                      tabIndex={-1}
                    />
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="PrivateNotchedOutline-legendLabelled-11"
                      >
                        <span>
                          Explications détaillées
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                className="textColor"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Justificatifs
                </p>
              </div>
              <div>
                Types de fichiers acceptés : JPEG, PNG et PDF 
              </div>
              <div>
                <label
                  className="sc-bczRLJ bMBuBa  "
                  htmlFor="file"
                  onClick={[Function]}
                >
                  <input
                    accept=".jpeg,.png,.pdf,.jpg"
                    multiple={true}
                    name="file"
                    onChange={[Function]}
                    onClick={[Function]}
                    type="file"
                  />
                  <div
                    className="MuiBox-root css-cksj3t"
                  >
                    <div
                      className="center"
                    >
                      <h3>
                        Glisser les pièces jointes pour ajouter des preuves justificatives
                      </h3>
                    </div>
                  </div>
                </label>
              </div>
              <div>
                <ul
                  className="afficherFichiers"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          className="boutonsFormulaire"
        >
          <button
            className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-1r7wqeh-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Annuler
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-122gtp7-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            id="boutonSoumettre"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            Soumettre
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
